### PR TITLE
Enabled TL as default for VM Simple Windows

### PR DESCRIPTION
**Pull request by AjKundnani**
# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

* Added Trusted Launch configuration into template.
* Removed Gen1 OS Image references from allowed values. Scoped to Gen2-TL capable OS images.

### CODE REVIEW

**Comment by azure-quickstarts** [comment]
@cynthn - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Comment by azure-quickstarts** [comment]
@cynthn - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Comment by azure-quickstarts** [comment]
@cynthn - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Comment by AjKundnani** [comment]
@bmoore-msft - The **BPA** warnings are from original version of template - storage account, it is not due to or caused by Trusted Launch configuration changes made as part of PR.

File: quickstarts/microsoft.compute/vm-simple-windows/main.bicep [hunk 1]
```diff
@@ -240,10 +231,31 @@ resource vm 'Microsoft.Compute/virtualMachines@2021-03-01' = {
...
+resource vmExtension 'Microsoft.Compute/virtualMachines/extensions@2022-03-01' = if (secureBoot) {
+  parent: vm
+  name: extensionName
+  location: location
+  properties: {
+    publisher: extensionPublisher
+    type: extensionName
+    typeHandlerVersion: extensionVersion
+    autoUpgradeMinorVersion: true
+    settings: {
```
Comment (Bab95) [inline quickstarts/microsoft.compute/vm-simple-windows/main.bicep:250]: now that GA extension support autoupgrade, please add autoupgrade property here as well.

"enableAutomaticUpgrade": true,

Comment (AjKundnani) [reply quickstarts/microsoft.compute/vm-simple-windows/main.bicep:251]: @Bab95 - Deployment pipeline are generating error for Public and Gov Cloud:

Status Message: VM Extension with publisher 'Microsoft.Azure.Security.WindowsAttestation' and type 'GuestAttestation' does not support setting enableAutomaticUpgrade property to true on this subscription. (Code:OperationNotAllowed)

**Comment by azure-quickstarts** [comment]
@cynthn - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

File: quickstarts/microsoft.compute/vm-simple-windows/createUiDefinition.json [hunk 2]
```diff
@@ -63,150 +63,42 @@
...
-                                    "label": "2012-datacenter-gensecond"
-                                },
-                                {
-                                    "value": "2012-Datacenter-smalldisk",
-                                    "label": "2012-Datacenter-smalldisk"
-                                },
-                                {
-                                    "value": "2012-datacenter-smalldisk-g2",
-                                    "label": "2012-datacenter-smalldisk-g2"
-                                },
```
Comment (sirfamelin) [inline quickstarts/microsoft.compute/vm-simple-windows/createUiDefinition.json:89]: Just curious, why are these removed? Comment applies everywhere a valid sku was removed

**Review by sirfamelin — COMMENTED** [review]
Why was azuredeploy.json deleted? Isn't that the main arm template?

File: quickstarts/microsoft.compute/vm-simple-windows/main.bicep [hunk 3]
```diff
@@ -100,15 +63,39 @@ param location string = resourceGroup().location
...
 var storageAccountName = 'bootdiags${uniqueString(resourceGroup().id)}'
 var nicName = 'myVMNic'
 var addressPrefix = '10.0.0.0/16'
 var subnetName = 'Subnet'
 var subnetPrefix = '10.0.0.0/24'
 var virtualNetworkName = 'MyVNET'
 var networkSecurityGroupName = 'default-NSG'
+var securityProfileJson = {
+  uefiSettings: {
+    secureBootEnabled: secureBoot
```
Comment (sirfamelin) [inline quickstarts/microsoft.compute/vm-simple-windows/main.bicep:88]: What is the reasoning behind paramterizing secureboot and not vtpm? Why not keep it consistent, either have both as params or don't expose either as params. I would prefer not having either as params so that sb and vtpm are on by default.

File: quickstarts/microsoft.compute/vm-simple-windows/main.bicep [hunk 4]
```diff
@@ -100,15 +63,39 @@ param location string = resourceGroup().location
...
+  'Standard'
+  'TrustedLaunch'
+])
+param securityType string = 'TrustedLaunch'
+
+@description('Secure Boot setting of the virtual machine. Applicable for securityType TrustedLaunch.')
+param secureBoot bool = true
+
+@description('Custom Attestation Endpoint to attest to. By default, MAA and ASC endpoints are empty and Azure values are populated based on the location of the VM.')
+param maaEndpoint string = ''
```
Comment (sirfamelin) [inline quickstarts/microsoft.compute/vm-simple-windows/main.bicep:77]: I don't remember off the top of my head, but can you check to see if it is possible to deploy the extension without specifying any endpoint in the request?

**Comment by AjKundnani** [comment]
> Why was azuredeploy.json deleted? Isn't that the main arm template?

The azureDeploy.json will get auto-generated by pipeline after PR is merged.

File: quickstarts/microsoft.compute/vm-simple-windows/createUiDefinition.json (hunk 2 above)
Comment (AjKundnani) [reply quickstarts/microsoft.compute/vm-simple-windows/createUiDefinition.json:89]: Trusted Launch is not supported for Windows Server 2012/2012 R2.
It is only supported for Gen2 OS images. Hence all Gen1 OS images were removed.

File: quickstarts/microsoft.compute/vm-simple-windows/main.bicep (hunk 3 above)
Comment (AjKundnani) [reply quickstarts/microsoft.compute/vm-simple-windows/main.bicep:88]: Turning On Secure boot has impact on bootchain as it will prohibit VM boot if boot chain is not attested. Providing parameter to users if they need to turn it OFF in their local copy of template.
vTPM is additional functionality and pre-requisite for modern OS like Windows 11. It does not impacts the boot process of VM.

**Comment by azure-quickstarts** [comment]
@cynthn - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Comment by azure-quickstarts** [comment]
@cynthn - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

File: quickstarts/microsoft.compute/vm-simple-windows/main.bicep (hunk 4 above)
Comment (AjKundnani) [reply quickstarts/microsoft.compute/vm-simple-windows/main.bicep:77]: The deployment succeeds in absence of the entire settings:{} block. However unable to view the attestation status in Portal if deployed without maa endpoint details.

**Deployed**:
image

**Expected**
image

**Comment by azure-quickstarts** [comment]
@cynthn - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Comment by azure-quickstarts** [comment]
@cynthn - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

File: quickstarts/microsoft.compute/vm-simple-windows/createUiDefinition.json (hunk 2 above)
Comment (sirfamelin) [reply quickstarts/microsoft.compute/vm-simple-windows/createUiDefinition.json:89]: I was referring to the valid G2 skus. I see that there are G2 (gen 2) skus removed.

File: quickstarts/microsoft.compute/vm-simple-windows/main.bicep (hunk 3 above)
Comment (sirfamelin) [reply quickstarts/microsoft.compute/vm-simple-windows/main.bicep:88]: The user can just set the secureBootEnabled to false manually in their local copy. By parameterizing it will just add one more thing for the user to decide.

On a side note, secureboot will prohibit a VM from booting if there are unsigned kernel/modules in the boot chain but it does not rely on attestation.

File: quickstarts/microsoft.compute/vm-simple-windows/main.bicep (hunk 4 above)
Comment (sirfamelin) [reply quickstarts/microsoft.compute/vm-simple-windows/main.bicep:77]: OK, in that case can you give the users a set list of endpoints to choose from? If it is freeform, user can enter an incorrect endpoint.

**Comment by azure-quickstarts** [comment]
@cynthn - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

File: quickstarts/microsoft.compute/vm-simple-windows/main.bicep (hunk 3 above)
Comment (AjKundnani) [reply quickstarts/microsoft.compute/vm-simple-windows/main.bicep]: @sirfamelin - Replaced secureBoot parameter. Kindly review, also review the guest attestation extension condition.

File: quickstarts/microsoft.compute/vm-simple-windows/main.bicep (hunk 4 above)
Comment (AjKundnani) [reply quickstarts/microsoft.compute/vm-simple-windows/main.bicep:92]: @sirfamelin - Added allowed values as discussed, kindly review.

**Comment by azure-quickstarts** [comment]
@cynthn - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Review by sirfamelin — APPROVED** [review]
LGTM
